runners/reference/job-server/src/test/java/org/apache/beam/runners/reference/job/ReferenceRunnerJobServiceTest.java

```diff
@@ -27,8 +47,54 @@
  */
 @RunWith(JUnit4.class)
 public class ReferenceRunnerJobServiceTest {
+  @Rule public TemporaryFolder temp = new TemporaryFolder();
+
+  private InProcessServerFactory serverFactory = InProcessServerFactory.create();
+  private ReferenceRunnerJobService service;
+  private GrpcFnServer<ReferenceRunnerJobService> server;
+  private JobServiceBlockingStub stub;
+
+  @Before
+  public void setup() throws Exception {
+    service = ReferenceRunnerJobService.create(serverFactory);
+    server = GrpcFnServer.allocatePortAndCreateFor(service, serverFactory);
+    stub =
+        JobServiceGrpc.newBlockingStub(
+            InProcessChannelBuilder.forName(server.getApiServiceDescriptor().getUrl()).build());
+  }
+
+  @After
+  public void teardown() throws Exception {
+    server.close();
+  }
+
   @Test
-  public void testPrepareJob() {
-    // TODO: Implement when PrepareJob is implemented.
+  public void testPrepareJob() throws Exception {
+    PrepareJobResponse response =
+        stub.prepare(
+            PrepareJobRequest.newBuilder()
+                .setPipelineOptions(Struct.getDefaultInstance())
+                .setPipeline(Pipeline.getDefaultInstance())
+                .setJobName("myJobName")
+                .build());
+
+    ApiServiceDescriptor stagingEndpoint = response.getArtifactStagingEndpoint();
+    ArtifactServiceStager stager =
+        ArtifactServiceStager.overChannel(
+            InProcessChannelBuilder.forName(stagingEndpoint.getUrl()).build());
+    File foo = writeTempFile("foo", "foo, bar, baz".getBytes());
+    File bar = writeTempFile("spam", "spam, ham, eggs".getBytes());
+    stager.stage(ImmutableList.<File>of(foo, bar));
+    // TODO: 'run' the job with some sort of noop backend, to verify state is cleaned up.
+    // TODO: Verify that the artifacts have been staged
```

Comment:
Filling in this TODO about verifying the staged artifacts should be a high priority PR before you move onto something else.

Reply:
ack